pkg/deploy/predeploy.go

```diff
@@ -249,3 +240,12 @@ func (d *deployer) ensureMonitoringCertificates(ctx context.Context, serviceKeyV
 
 	return nil
 }
+
+func (d *deployer) getRpServicePrincipalID(ctx context.Context) (string, error) {
+	deployment, err := d.deployments.Get(ctx, d.config.ResourceGroupName, rpManagedIdentityDeploymentName)
```

Comment:
almost worth dropping Outputs entirely and calling applications.GetServicePrincipalsIDByAppID here.

Reply:
agreed.